activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeBPMNTimerIT.java

```diff
@@ -0,0 +1,292 @@
+import org.activiti.api.process.model.builders.ProcessPayloadBuilder;
+import org.activiti.api.process.model.events.BPMNTimerCanceledEvent;
+import org.activiti.api.process.model.events.BPMNTimerEvent;
+import org.activiti.api.process.model.events.BPMNTimerFiredEvent;
+import org.activiti.api.process.model.events.BPMNTimerScheduledEvent;
+import org.activiti.api.process.runtime.ProcessRuntime;
+import org.activiti.api.runtime.shared.query.Page;
+import org.activiti.api.runtime.shared.query.Pageable;
+import org.activiti.api.task.model.Task;
+import org.activiti.api.task.model.builders.TaskPayloadBuilder;
+import org.activiti.api.task.runtime.TaskRuntime;
+import org.activiti.engine.ManagementService;
+import org.activiti.engine.ProcessEngineConfiguration;
+import org.activiti.engine.impl.test.JobTestHelper;
+import org.activiti.spring.boot.process.listener.DummyBPMNTimerCanceledListener;
+import org.activiti.spring.boot.process.listener.DummyBPMNTimerExecutionSuccessFiredListener;
+import org.activiti.spring.boot.process.listener.DummyBPMNTimerFiredListener;
+import org.activiti.spring.boot.process.listener.DummyBPMNTimerScheduledListener;
+import org.activiti.spring.boot.security.util.SecurityUtil;
+import org.activiti.spring.boot.test.util.ProcessCleanUpUtil;
+import org.assertj.core.groups.Tuple;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+public class ProcessRuntimeBPMNTimerIT {
+    
+    private static final String PROCESS_INTERMEDIATE_TIMER_EVENT = "intermediateTimerEventExample";
+    private static final String PROCESS_TIMER_CANCELED_EVENT = "testTimerCancelledEvent";
+ 
+    @Autowired
+    private ProcessRuntime processRuntime;
+    
+    @Autowired
+    private TaskRuntime taskRuntime;
+
+    @Autowired
+    private SecurityUtil securityUtil;
+
+    @Autowired
+    private DummyBPMNTimerFiredListener listenerFired;
+    
+    @Autowired
+    private DummyBPMNTimerScheduledListener listenerScheduled;
+    
+    @Autowired
+    private DummyBPMNTimerCanceledListener listenerCanceled;
+
+    @Autowired
+    private  DummyBPMNTimerExecutionSuccessFiredListener listenerExecutionSuccess;
+    
+    @Autowired
+    private ProcessCleanUpUtil processCleanUpUtil;
+    
+    @Autowired
+    private ManagementService managementService;
+    
+    @Autowired
+    ProcessEngineConfiguration processEngineConfiguration;
+    
+    @Before
+    public void setUp() {
+        clear();
+        
+    }
+
+    @After
+    public void tearDown() {
+        processCleanUpUtil.cleanUpWithAdmin();
+    }
+
+    @Test
+    public void shouldGetTimerFiredScheduledEventsForProcessWithTimer() {
```

Comment:
maybe the first scenario can be splitted in two different scenarios, one without delay and one with delay?  That might help readability?